google-cloud-storage/src/test/java/com/google/cloud/storage/ByteRangeSpecTest.java

```diff
@@ -441,129 +444,168 @@ private static void threeWayEqual(
   @RunWith(Parameterized.class)
   public static final class RangeScenarios {
 
-    private static final long INF = Long.MAX_VALUE;
     private final RangeScenario rs;
+    private final RangeScenario.Expectations expect;
 
     public RangeScenarios(RangeScenario rs) {
       this.rs = rs;
+      this.expect = rs.getExpectations();
     }
 
     @Test
     public void httpRangeHeader() {
-      assertThat(rs.getSpec().getHttpRangeHeader()).isEqualTo(rs.getExpectedHttpRange());
+      assertThat(rs.getSpec().getHttpRangeHeader()).isEqualTo(expect.getHttpRange());
     }
 
     @Test
     public void beginOffset() {
-      assertThat(rs.getSpec().beginOffset()).isEqualTo(rs.getExpectedBeginOffset());
+      assertThat(rs.getSpec().beginOffset()).isEqualTo(expect.getBeginOffset());
     }
 
     @Test
     public void endOffset() {
-      assertThat(rs.getSpec().endOffset()).isEqualTo(rs.getExpectedEndOffset());
+      assertThat(rs.getSpec().endOffset()).isEqualTo(expect.getEndOffset());
+    }
+
+    @Test
+    public void endOffsetInclusive() {
+      assertThat(rs.getSpec().endOffsetInclusive()).isEqualTo(expect.getEndOffsetInclusive());
     }
 
     @Test
     public void length() {
-      assertThat(rs.getSpec().length()).isEqualTo(rs.getExpectedLength());
+      assertThat(rs.getSpec().length()).isEqualTo(expect.getLength());
     }
 
     @Parameters(name = "{0}")
     public static Iterable<Object[]> testCases() {
+      // expect that by default, a range should be from zero to infinity
       Stream<RangeScenario> bothNullOrEmpty =
-          Stream.of(
+          RangeScenario.expectThat()
+              .beginOffset(0L)
+              .endOffset(EFFECTIVE_INFINITY)
+              .endOffsetInclusive(EFFECTIVE_INFINITY)
+              .length(EFFECTIVE_INFINITY)
+              .httpRange(null)
+              .isApplicableTo(
```

Comment:
The new builder which allows this more explicit description of the tests is the majority of the change in this PR.